modules/gdscript/gdscript_compiler.cpp

```diff
@@ -2647,7 +2647,10 @@ Error GDScriptCompiler::_prepare_compilation(GDScript *p_script, const GDScriptP
 
 	GDScriptDataType base_type = _gdtype_from_datatype(p_class->base_type, p_script, false);
 
+	ERR_FAIL_COND_V_MSG(base_type.native_type == StringName(), ERR_BUG, vformat(R"(Failed to get base class for "%s")", p_script->path));
```

Comment:
Should this be an engine-level error (`ERR_BUG`), or a user-level error (`_set_error` + returning the most relevant error code)?

Reply:
I think it's fine, this should not happen as a result of user error, it's an analyzer/compiler bug.